packages/angular_devkit/schematics/src/engine/engine.ts

```diff
@@ -76,16 +83,38 @@ export class SchematicEngine<CollectionT extends object, SchematicT extends obje
       return collection;
     }
 
+    const [description, bases] = this._createCollectionDescription(name);
+
+    collection = new CollectionImpl<CollectionT, SchematicT>(description, this, bases);
+    this._collectionCache.set(name, collection);
+    this._schematicCache.set(name, new Map());
+
+    return collection;
+  }
+
+  private _createCollectionDescription(
+    name: string,
+    parentNames?: Set<string>,
```

Comment:
Just one nit but not stopping the approval; `parentNames = new Set<string>()` then lower just add regardless of whether it's defined or not.